WordPress/Classes/ViewRelated/Reader/DetailWebView/ReaderDetailToolbar.swift

```diff
@@ -0,0 +1,323 @@
+            let viewController = viewController as? UIViewController & UIViewControllerTransitioningDelegate else {
+            return
+        }
+
+        if !readerPost.isSavedForLater {
+            FancyAlertViewController.presentReaderSavedPostsAlertControllerIfNecessary(from: viewController)
+        }
+
+        ReaderSaveForLaterAction().execute(with: readerPost, context: context, origin: .postDetail) { [weak self] in
+            self?.saveForLaterButton.isSelected = readerPost.isSavedForLater
+            self?.prepareActionButtonsForVoiceOver()
+        }
+    }
+
+    @IBAction func didTapReblog(_ sender: Any) {
+        guard let post = post, let viewController = viewController else {
+            return
+        }
+
+        ReaderReblogAction().execute(readerPost: post, origin: viewController, reblogSource: .detail)
+    }
+
+    @IBAction func didTapComment(_ sender: Any) {
+        guard let post = self.post, let viewController = viewController else {
+            return
+        }
+
+        ReaderCommentAction().execute(post: post, origin: viewController)
+    }
+
+    @IBAction func didTapLike(_ sender: Any) {
+        guard let post = post else {
+            return
+        }
+
+        if !post.isLiked {
+            UINotificationFeedbackGenerator().notificationOccurred(.success)
+        }
+
+        let service = ReaderPostService(managedObjectContext: post.managedObjectContext!)
+        service.toggleLiked(for: post, success: nil, failure: { [weak self] (error: Error?) in
+            self?.trackArticleDetailsLikedOrUnliked()
+            if let anError = error {
+                DDLogError("Error (un)liking post: \(anError.localizedDescription)")
+            }
+        })
+    }
+
+    // MARK: - Styles
+
+    private func applyStyles() {
+        backgroundColor = .listForeground
+        dividerView.backgroundColor = .divider
+
+        WPStyleGuide.applyReaderCardActionButtonStyle(commentButton)
+        WPStyleGuide.applyReaderCardActionButtonStyle(likeButton)
+    }
+
+    // MARK: - Configuration
+
+    private func configureActionButtons() {
+        resetActionButton(likeButton)
+        resetActionButton(commentButton)
+        resetActionButton(saveForLaterButton)
+        resetActionButton(reblogButton)
+
+        guard let post = post else {
+            assertionFailure()
+            return
+        }
+
+        // Show likes if logged in, or if likes exist, but not if external
+        if (ReaderHelpers.isLoggedIn() || post.likeCount.intValue > 0) && !post.isExternal {
+            configureLikeActionButton()
+        }
+
+        // Show comments if logged in and comments are enabled, or if comments exist.
+        // But only if it is from wpcom (jetpack and external is not yet supported).
+        // Nesting this conditional cos it seems clearer that way
+        if (post.isWPCom || post.isJetpack) && !shouldHideComments {
```

Comment:
The only place we set `shouldHideComments` is in https://github.com/wordpress-mobile/WordPress-iOS/blob/0bdc8da476a29b1574d17e6a027783ffa7e46d40/WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m#L1182 but it checks the variable setting in `ReaderDetailViewController`

Reply:
This is for the future. :)

In a subsequent branch I added that: https://github.com/wordpress-mobile/WordPress-iOS/blob/issue/14252_new_reader_detail_missing_features/WordPress/Classes/ViewRelated/Reader/ReaderDetailWebviewViewController.swift#L58

Reply:
And you'll set it to true where needed as well?

Reply:
Yes! Also here: https://github.com/wordpress-mobile/WordPress-iOS/blob/6f792020947308b8053672488eff1a91c1aaa141/WordPress/Classes/ViewRelated/Reader/ReaderCommentsViewController.m#L1183 :)